src/components/providers/BackgroundTasksProvider.tsx

```diff
@@ -0,0 +1,93 @@
+import { createContext, useContext, useEffect, useState } from "react";
+import BackgroundFetch from "react-native-background-fetch";
+
+type ContextState = {
+  isTrackingEnabled: boolean;
+  setTrackingEnabled?: (value: boolean) => void;
+};
+
+const initialState: ContextState = {
+  isTrackingEnabled: false,
+};
+
+const BackgroundTasksContext = createContext<ContextState>(initialState);
+
+const BackgroundTasksProvider: React.FC<{ children: React.ReactNode }> = ({
+  children,
+}) => {
+  const [isTrackingEnabled, setTrackingEnabled] = useState<boolean>(false);
+
+  useEffect(() => {
+    initBackgroundFetch();
```

Comment:
is this starting the background? if so there should be a conditional to only start it when this flag isTrackingEnabled is true

Reply:
this just configures it

Reply:
Gotcha! I thought this was initiate of initialize because It had the setTrackingEnabled(true) inside